File: abstract-factory/c-sharp/OracleDatabaseFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AbstractFactory
{
    // Oracle için ilişkili nesneler üretildi.
    class OracleDatabaseFactory : IDatabaseFactory
    {
        public Command CreateCommand()
        {
            return new OracleCommand();
        }
        public Connection CreateConnection()
        {
            return new OracleConnection();
        }
    }
}

File: abstract-factory/go/oracle-database-factory.go
package main

// Oracle için ilişkili nesneler üretildi.
type oracleDatabaseFactory struct{}

func NewOracleDatabaseFactory() *oracleDatabaseFactory {
	return &oracleDatabaseFactory{}
}

func (f *oracleDatabaseFactory) createConnection() Connection {
	return newOracleConnection()
}

func (f *oracleDatabaseFactory) createCommand() Command {
	return newOracleCommand()
}
